src/main/java/Duke.java

```diff
@@ -1,10 +1,180 @@
+    }
+
+    private void greet() {
+        System.out.println(GREET_MESSAGE);
+        printDivider();
+    }
+
+    private void exit() {
+        System.out.println(EXIT_MESSAGE);
+        printDivider();
+    }
+
+    private void echo(String input) {
+        System.out.println(input);
+        printDivider();
+    }
+
+    private void showInvalidCommandMessage() {
+        System.out.println(INVALID_COMMAND_MESSAGE);
+        printDivider();
+    }
+
+    private void showHelpMessage() {
+        System.out.println(HELP_MESSAGE);
+        printDivider();
+    }
+
+    private void addTodo(String taskDescription) {
+        Task addedTask = TASK_MANAGER.addTodo(taskDescription);
+        System.out.println(ADD_TASK_MESSAGE + "\n" + addedTask.toString());
+        printTasksCount();
+        printDivider();
+    }
+
+    private void addDeadline(String taskDescription, String taskDue) {
+        Task addedTask = TASK_MANAGER.addDeadline(taskDescription, taskDue);
+        System.out.println(ADD_TASK_MESSAGE + "\n" + addedTask.toString());
+        printTasksCount();
+        printDivider();
+    }
+
+    private void addEvent(String taskDescription, String eventDateTime) {
+        Task addedTask = TASK_MANAGER.addEvent(taskDescription, eventDateTime);
+        System.out.println(ADD_TASK_MESSAGE + "\n" + addedTask.toString());
+        printTasksCount();
+        printDivider();
+    }
+
+    private void listTasks() {
+        System.out.println(LIST_TASKS_MESSAGE);
+        TASK_MANAGER.listTasks();
+        printDivider();
+    }
+
+    private void clearAllTasks() {
+        TASK_MANAGER.clearAllTasks();
+        System.out.println(CLEAR_TASKS_MESSAGE);
+        printDivider();
+    }
+
+    private void markTaskAsDone(int taskIndex) {
+        Task finishedTask = TASK_MANAGER.markTaskAsDone(taskIndex);
+        System.out.println(MARK_TASK_AS_DONE_MESSAGE + "\n" + finishedTask.toString());
+        printDivider();
+    }
+
+    public void execute() {
+
+        String userInputString;
+        String[] parsedUserInput;
+        String command;
+        String[] params;
+
+        greet();
+        while (true) {
+            userInputString = SCANNER.nextLine();
+
+            if (userInputString.equalsIgnoreCase(COMMAND_EXIT)) {
+                exit();
+                break;
+            }
+            if (userInputString.equalsIgnoreCase(COMMAND_LIST_TASKS)) {
+                listTasks();
+                continue;
+            }
+            if (userInputString.equalsIgnoreCase(COMMAND_CLEAR_TASKS)) {
+                clearAllTasks();
+                continue;
+            }
+            if (userInputString.equalsIgnoreCase(COMMAND_HELP)) {
+                showHelpMessage();
+                continue;
+            }
+
+            parsedUserInput = PARSER.separateCommand(userInputString);
+            command = parsedUserInput[0].toLowerCase();
+
+            switch (command) {
```

Comment:
Could maybe add the switch block into a handleCommand function for better readability